alerts/cluster-monitoring-operator/TelemeterClientFailures.md

```diff
@@ -0,0 +1,75 @@
+# TelemeterClientFailures
+
+## Meaning
+
+The alert `TelemeterClientFailures` is triggered when the Telemeter client fails
+to send Telemetry data at a certain rate over a period of time
+to Red Hat.
+
+The `telemeter-client-xxxxxxxxx-xxxxx` pod running in the `openshift-monitoring`
+namespace collects [selected platform metrics](https://docs.openshift.com/container-platform/latest/support/remote_health_monitoring/showing-data-collected-by-remote-health-monitoring.html#showing-data-collected-from-the-cluster_showing-data-collected-by-remote-health-monitoring)
+from the `prometheus-k8s` service at
+regular intervals using the `/federate` endpoint and ships them periodically
+to Red Hat using a custom secured protocol.
+
+## Impact
+
+When the alert fires, Red Hat support and engineering teams don't have a complete
+view of the cluster anymore. It may hinder the ability for Red Hat to
+proactively detect issues in the cluster.
+
+## Diagnosis
+
+Review the logs for the pod `telemeter-client-xxxxxxxxx-xxxxx`
+in the `openshift-monitoring` namespace
+
+You can review the logs for the `telemeter-client-xxxxxxxxx-xxxxx` pod in the
+`openshift-monitoring` namespace by running the following command:
+
+```console
+$ oc logs -n openshift-monitoring deployment.apps/telemeter-client -c telemeter-client | less
+```
+
+## Mitigation
+
+* From `4.16` there is metric `metricsclient_http_requests_total` which can be
```

Comment:
`metricsclient_http_requests_total`is since 4.17
```
$ oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.17.0-0.nightly-2024-08-13-031847   True        False         7h23m   Cluster version is 4.17.0-0.nightly-2024-08-13-031847

$ token=`oc create token prometheus-k8s -n openshift-monitoring`
$ oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $token" 'https://thanos-querier.openshift-monitoring.svc:9091/api/v1/label/__name__/values' | jq | grep metricsclient
    "metricsclient_http_requests_total",
    "metricsclient_request_retrieve",
    "metricsclient_request_send",
```
not in 4.16
```
$ oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.16.0-0.nightly-2024-08-14-124952   True        False         5h46m   Cluster version is 4.16.0-0.nightly-2024-08-14-124952
$ token=`oc create token prometheus-k8s -n openshift-monitoring`
$ oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $token" 'https://thanos-querier.openshift-monitoring.svc:9091/api/v1/label/__name__/values' | jq | grep metricsclient
    "metricsclient_request_retrieve",
    "metricsclient_request_send",
```

Reply:
right my bad :)